### PR TITLE
return an array with only prime numbers

### PR DESCRIPTION
**Pull request by saumya665**

### CODE REVIEW

File: curriculum/js2/3.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/**
+ * Given an array, return an array of the same length
+ * where all elements that are not prime is changed to 1
+ * */
+
+ const prime=(num,i=2)=>{
+   if(num<=i){
```
Comment (MadhuBeesetty) [inline curriculum/js2/3.js]: why not "==" ??

File: curriculum/js2/3.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      return true;
+    }
+    if(num%i==0){
+      return false;
+    }
+    return prime(num,i+1);
+  }
+
```
Comment (MadhuBeesetty) [inline curriculum/js2/3.js]: why two separate functions?

File: curriculum/js2/3.js (hunk 1 above)
Comment (saumya665) [reply curriculum/js2/3.js]: == gives a call stack exceeded so used <=

File: curriculum/js2/3.js (hunk 2 above)
Comment (saumya665) [reply curriculum/js2/3.js]: makes it easier to debug

File: curriculum/js2/3.js (hunk 1 above)
Comment (dsope05) [reply curriculum/js2/3.js]: the test passes in 1, so if you use '==' it will never be true

**Pull request merged**